### PR TITLE
Make sndfile.h independent of build system variables

### PR DESCRIPTION
**Pull request by manxorist**
Fixes <https://github.com/libsndfile/libsndfile/issues/868>.

### CODE REVIEW

**Comment by SoapGentoo** [comment]
Thanks, I wanted to get around to doing this myself too. The only downside (not for me!) is that some people will cry that their that MSVC 2008 support is lost with this. At this point, I wouldn't care about MSVC 2008 anymore, and people who want to use it should just add some macro hacks themselves.

**Review by SoapGentoo — APPROVED** [review]

**Comment by manxorist** [comment]
>  The only downside (not for me!) is that some people will cry that their that MSVC 2008 support is lost with this.

As I noted in #868, VS2008 (which does not provide <stdint.h>) was already broken 9 years ago. I guess people would have complained by now. So this should really be a non-issue.

**Comment by manxorist** [comment]
I'll look into the CI failures and update the pull request once I have figured it out.

**Comment by SoapGentoo** [comment]
right, I didn't connect the dots. You will likely just have to fix that pedantic header test

**Comment by evpobr** [comment]
Hi @manxorist .

**Comment by evpobr** [comment]
@SoapGentoo , will you take this PR?

> The only downside (not for me!) is that some people will cry that their MSVC 2008 support is dropped with this. At this point, I wouldn't care about MSVC 2008 anymore, and people who want to use it should just add some macro hacks themselves.

We require C99-compliant compiler in `README.md`.

File: docs/index.md [hunk 1]
```diff
@@ -138,7 +138,7 @@ Here is the release history for libsndfile:
   Linux platforms. Other numerous fixes to CMake build system, consider it
   stable now. Fix some memory leaks. Fix handling of some SD2 files. Update
   documentation. Integrate GitHub Actions for faster test builds and Oss-Fuzz
-  for fuzzing tests. Move sndfile.h.in from src/ to include/ directory. To avoid
+  for fuzzing tests. Move sndfile.h from src/ to include/ directory. To avoid
```
Comment (evpobr) [inline docs/index.md:141]: Hmm, it is changelog of 1.0.30, do we really need to rewrite it?

**Review by evpobr — APPROVED** [review]

**Comment by evpobr** [comment]
Thanks!

File: include/sndfile.h [hunk 2]
```diff
@@ -365,9 +365,9 @@ typedef	struct sf_private_tag	SNDFILE ;
 ** and the Microsoft compiler.
 */
 
-typedef @TYPEOF_SF_COUNT_T@	sf_count_t ;
+typedef int64_t			sf_count_t ;
 #ifndef SF_COUNT_MAX
-#define SF_COUNT_MAX		@SF_COUNT_MAX@
+#define SF_COUNT_MAX		0x7FFFFFFFFFFFFFFFLL
```
Comment (SoapGentoo) [inline include/sndfile.h:370]: @manxorist I thought a bit more, what about `INT64_MAX` here? That avoids the `long` vs `long long` literal discrepancy

Comment (manxorist) [reply include/sndfile.h:370]: Yeah, that's probably a good additional cleanup. `LL`/`ll` suffix was what was always used before, I did just not consider changing it.

Do you want me to submit another pull request or will you just commit that yourself?

Comment (SoapGentoo) [reply include/sndfile.h:370]: I'm a bit busy, so if you did it, that would help me :slightly_smiling_face:

Comment (manxorist) [reply include/sndfile.h:370]: I'm also busy right now, I'll do it tomorrow. 🙂

Comment (manxorist) [reply include/sndfile.h:370]: found some spare time, see #870

**Pull request merged**